velox/serializers/PrestoSerializer.cpp

```diff
@@ -826,7 +826,7 @@ std::string readLengthPrefixedString(ByteStream* source) {
   return value;
 }
 
-void checkTypeEncoding(std::string encoding, TypePtr type) {
+void checkTypeEncoding(std::string_view encoding, const TypePtr& type) {
```

Comment:
nit: Maybe `const std::string&` is better than `std::string_view`?

I noticed that `typeToEncodingName()` below will return a `std::string`, and there will be an `encoding == kindEncoding` check.

Since `operator==(std::string_view, std::string)` does not exist, it means that `std::string_view == std::string` requires an implicit conversion?(probably at compile time)

https://en.cppreference.com/w/cpp/string/basic_string_view/operator_cmp